File: wrap_functions.js
const fs = require('fs');

let content = fs.readFileSync('code.gs', 'utf8');

const funcs = [
    "updateSettings",
    "editUserRole",
    "deleteUserRole",
    "addUserRole",
    "assignSubToPeriod",
    "cancelMySubDuty",
    "updateAbsence",
    "submitAbsence",
    "cancelAbsence"
];

funcs.forEach(funcName => {
    // We are looking for the function definition: function submitAbsence(formData) { ... }
    const regex = new RegExp(`function \\s+${funcName}\\s*\\([^)]*\\)\\s*{[\\s\\S]*?catch\\s*\\(([^)]+)\\)\\s*{`, 'g');

    content = content.replace(regex, (match, errVar) => {
        // If it already has notifyAdminOfError, skip
        if (match.includes('notifyAdminOfError')) return match;

        return `${match}\n    notifyAdminOfError("${funcName}", ${errVar});`;
    });
});

fs.writeFileSync('code.gs', content, 'utf8');

File: wrap_functions.py
import re

def main():
    with open('code.gs', 'r') as f:
        content = f.read()

    # Functions to wrap
    funcs = [
        "updateSettings",
        "editUserRole",
        "deleteUserRole",
        "addUserRole",
        "assignSubToPeriod",
        "cancelMySubDuty",
        "updateAbsence",
        "submitAbsence",
        "cancelAbsence"
    ]

    # We will look for the catch block of each function if they exist and add notifyAdminOfError.
    # Most of these functions already have try/catch blocks that return `{success: false, error: ...}`.
    # Let's find them.

    for func in funcs:
        # Simple regex to find catch blocks inside these functions
        # This is a bit tricky because we want to inject it inside the catch block.
        # Let's look for `catch (e)` or `catch (error)` or `catch(e)`
        pass # Actually, manual replacement might be safer to ensure we get the right ones.

    with open('code.gs', 'w') as f:
        f.write(content)

if __name__ == "__main__":
    main()
